cmd/clusterctl/client/get_kubeconfig_test.go

```diff
@@ -48,9 +56,57 @@ func Test_clusterctlClient_GetKubeconfig(t *testing.T) {
 		{
 			name:      "returns error if unable namespace is empty",
 			client:    badClient,
-			options:   GetKubeconfigOptions{Kubeconfig: Kubeconfig(kubeconfig)},
+			options:   GetKubeconfigOptions{Kubeconfig: Kubeconfig(kubeconfig), UserKubeconfig: true},
 			expectErr: true,
 		},
+		{
+			name: "returns the default user kubeconfig",
+			client: func() *fakeClient {
+				newclusterClient := newFakeCluster(kubeconfig, configClient).
+					WithObjs(append(objects, getKubeconfigSecret(true))...)
+				return newFakeClient(configClient).WithCluster(newclusterClient)
+			}(),
+			options: GetKubeconfigOptions{
+				Kubeconfig:          Kubeconfig(kubeconfig),
+				Namespace:           "default",
+				WorkloadClusterName: "mgmt-cluster",
```

Comment:
```suggestion
				WorkloadClusterName: "workload-cluster",
```
nit: Using `mgmt-cluster` as the name of the workload cluster could be misleading. 
(Please adjust at other places as well 🙂)